### PR TITLE
Use connection ids instead of names

### LINKED ISSUE #3333: Allow editing a connection nickname
This field needs to be editable

image

We need to make sure that, upon changing a connection's nickname, all instances of it are reactively updated and no navigation is broken. This may require some structural changes in lower-level code.

### PR DESCRIPTION
**Pull request by seancolsen**
Related to #3328

## This PR:

- Fixes a bug where the user is unable to create new schemas.

    This bug stems from #3349 where we made some changes to the schemas API when listing schemas and creating a new schema.

    - In `develop`, creating a new schema is currently broken due to breaking API changes that we failed to handle. Whoops!
    
        This PR fixes that bug in 9feaca32c0928069aea042a93663ecdce4d2fdea by changing the frontend code to follow the modified API.

    - Anish made the other API changes in a backwards-compatible manner, so they do not affect functionality in `develop`.

- Changes the route for the Database Page and all child pages

    - Old: `/db/<connection_nickname>/`
    - New: `/db/<connection_id>/`

- Modifies the data structures used for the front end stores to reference connections by their ids instead of by their names. The purpose of this change is to lay groundwork for #3333.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the `develop` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by Anish9901** [comment]
No additional backend changes should be necessary @seancolsen, you're gtg!

**Comment by seancolsen** [comment]
Thanks @Anish9901!

File: mathesar/views.py [hunk 1]
```diff
@@ -182,13 +182,14 @@ def get_common_data(request, database=None, schema=None):
...
+def get_current_database(request, connection_id):
     """Get database from passed name, with fall back behavior."""
     successful_dbs, failed_dbs = _get_permissible_db_queryset(request)
     permitted_databases = successful_dbs | failed_dbs
-    if db_name is not None:
-        current_database = get_object_or_404(permitted_databases, name=db_name)
+    if connection_id is not None:
+        current_database = get_object_or_404(permitted_databases, id=connection_id)
     else:
+        # TODO: refactor connection names to ids
```
Comment (pavish) [inline mathesar/views.py:192]: I do not think this will break the demo setup. I'd like @mathemancer to take a look and confirm once, post merge.

**Review by pavish — APPROVED** [review]
@seancolsen Looks good to me!

I added a commit https://github.com/mathesar-foundation/mathesar/pull/3341/commits/a3c030fc94bf13e399aebca51a8aa17d8175c9e1 to fix formatting errors that were identified after the recent prettier fix in develop.

File: mathesar/views.py (hunk 1 above)
Comment (pavish) [reply mathesar/views.py:192]: @seancolsen I hope you'll followup regarding this.

Comment (seancolsen) [reply mathesar/views.py:192]: I removed the code comment in #3354.

**Pull request merged**